### ISSUE TITLE
Add backend modul to ease editor work

### ISSUE THREAD

**Issue by albig** [kitodo/kitodo-presentation#597]
## Description

Up to Kitodo.Presentation 2.x we had a very basic TYPO3 backend module. It was possible to 
* index single documents
* re-index one or more collections

The functionality was the same as the CLI jobs. But the big difference is, that you can give TYPO3 backend users (editors) the rights to use this module. But you cannot enable editors to use the CLI jobs.

**This module should be added again to make the daily work with Kitodo.Presentation easier and smoother. It should be the central point for editors and administrators.**

The following functions should be provided:
* index single documents (by ID, by URL) - same as CLI-job
* re-index one or more collections or all documents - same as CLI-job - this must be done asynchronously as it could take hours/days/weeks
* show Solr status information - like already done on edition solr record
* show status information about documents, collections
* list all collections and let user open records to edit description / images; allow to dele
* search for kitodo documents and provide preview-link (opens workview of document in another tab)
* ... keep it extendable for future ideas

The backend module should be written with Extbase/Fluid using Bootstrap which is already present in TYPO3 backend for this purpose.

## Expected benefits of this development

* Administrators
   * Transfer more duties to fellow users without admin rights
   * Faster support because no access to the console is necessary in most cases
* Editors
   * Easier to understand. Kitodo.Presentation has all on one dashboard.
   * Solve problems with index-updates independantly
   * 
* Marketing
   * eye-cather 

## Estimated Costs and Complexity

This is my rough estimation of costs in PT (day work):

* Add minimal backend module: 2-3 PT
* Add status overviews: 1 PT
* Add collection and document listings: 1-3 PT
* Design adaptations: 1 PT

sum: 5-8 PT

It's a **mid range** development project.

## Related Issues

* details of implementation depends on #595

**Comment by sebastian-meyer** [comment]
Unlike the old module, (re-)indexing has to be done asynchronous, because of the rather long runtime. I suggest implementing an indexing task for the TYPO3 scheduler (see #454). The backend module could trigger this task and provide parameters (i. e. which document/collection to index).

**Comment by albig** [comment]
Good point. I remember this odd behaviour. Of course this must be done somehow asynchronously. The solr extension may be a starting point. It shows progress bar in the scheduler task list.
